server/controllers/index.js

```diff
@@ -120,6 +115,15 @@ router.get('/api/cards', auth, (req, res) => {
     const gameState = Room.getById(roomId).gameState;
     const cards = gameState.getUnplayedCardsByUsername(user.username);
     if (cards) {
+        if (gameState.mode === 'custom') {
+            cards.forEach(card => {
```

Comment:
this is capable of invoking `res.status(err.code)` n times. it's not possible to respond multiple times to the same HTTP request.

moreover, there's a race condition. when game mode is custom, it will fire off all these async functions, but it will also continue to execute `res.status(200)`.

Reply:
you should do something like:

```js
(gameState.mode === 'custom'
? Promise.all(
  cards.map(card => db.getCard(card.cardId)
    .then(other => ({
      ...card,
      url: other.dataValues.url
    }))))
: Promise.resolve(cards))
.then(
  cards => res.status(200).json(cards),
  err => res.status(err.code).json({ message: err.message ));
```

Reply:
also we shouldn't use err.code as our HTTP status code.  
I am assuming `db.getCard()` would return _database_ error codes, not HTTP status codes. and even if it happened to be HTTP: the client talking to our backend wants to know the status of our _backend_, not of other services it depends on.